src/Discord.Net.Commands/Info/CommandInfo.cs

```diff
@@ -176,12 +175,22 @@ public Task<ExecuteResult> ExecuteAsync(ICommandContext context, ParseResult par
             }
         }
 
-        private async Task ExecuteAsyncInternal(ICommandContext context, object[] args, IServiceProvider services)
+        private async Task<IResult> ExecuteAsyncInternal(ICommandContext context, object[] args, IServiceProvider services)
         {
             await Module.Service._cmdLogger.DebugAsync($"Executing {GetLogText(context)}").ConfigureAwait(false);
             try
             {
-                await _action(context, args, services).ConfigureAwait(false);
+                var task = _action(context, args, services);
+                if (task is Task<IResult> returnTask)
```

Comment:
It does work, but probably not for users using command builders. Whoops.

The reason why I needed to return Task<IResult> is because of ModuleClassBuilder's Callback code: it returns a Task<IResult>, and unfortuantely due to contravariance not working the way I wanted it to, it would never trip the condition.